### DISCUSSION TITLE
One client per EAP identity credentials | How to solve?

### DISCUSSION THREAD

**Discussion by zephyrbl** [strongswan/strongswan#1675] [Q&A] [answered]
Hi,

I have configured StrongSwan with Let's Encrypt certificate and EAP-MSCHAPv2 authentication but noticed that two different devices are able to connect simultaneously to the server using the same EAP credentials. How to avoid this?

Example, User 1 connects to the server using the Login1/Password1, User2 should be able to connect using only its own Login2/Password2 but not the Login1/Password1.

ipsec.conf:

config setup
  strictcrlpolicy=yes
  uniqueids=never

conn roadwarrior
  auto=add
  compress=no
  type=tunnel
  keyexchange=ikev2
  fragmentation=yes
  forceencaps=yes

  ike=chacha20poly1305-sha512-curve25519-prfsha512,aes256-sha256-prfsha256-modp2048,aes256gcm16-sha384-prfsha384-ecp384,aes256-sha1-modp1024,aes128-sha1-modp1024,3des-sha1-modp1024!
  esp=chacha20poly1305-sha512,aes256gcm16-ecp384,aes256-sha256,aes256-sha1,3des-sha1!

  dpdaction=clear
  dpddelay=900s
  rekey=no
  left=%any
  leftid=@domain-name
  leftcert=cert.pem
  leftsendcert=always
  leftsubnet=0.0.0.0/0
  right=%any
  rightid=%any
  rightauth=eap-mschapv2
  eap_identity=%any
  rightdns=Google DNS
  rightsourceip=10.10.10.10/16
  rightsendcert=never

ipsec.secrets:

@domain-name : RSA "privkey"
user1 : EAP "password1"

I would appreciate if someone can help with this question.

Thanks in advance!

**Comment by drcyberg** [comment]
Hi!
Simply change the subnet mask value from `10.10.10.10/16` to `10.10.10.10/32`.
From here on, it only assigns 1 IP address.

**Reply by zephyrbl** [reply]
Hi, 

I think this will change only the pool range but not the connection behaviour per one user.

**Reply by drcyberg** [reply]
Indeed, I use `freeradius` to solve this problem.
And even professional. :,D

**Comment by misch42** [comment]
1) Migrate ot the swanctl configuration format. swanctl is so much 
easier and powerfull that legacy ipsec.conf.

2)  You might check the <conn>.unique paramter of the swanctl confige. 
See: https://docs.strongswan.org/docs/5.9/swanctl/swanctlConf.html

3) Also the duplicheck plugin solves your problem, independent of the 
unique option. See: 
https://docs.strongswan.org/docs/5.9/plugins/duplicheck.html

**Reply by zephyrbl** [reply]
Hi, 

I tried to install this plugin but with no success. 

Could you, please, advise how to install it using ./configure?
Maybe you can share some example of the command how it should look like?

Thanks in advance!

**Reply by tobiasbrunner** [reply]
Please **don't use** the _duplicheck_  plugin, it generally doesn't do what people think it does (read [the documentation](https://docs.strongswan.org/docs/5.9/plugins/duplicheck.html), in particular the part about what happens to the new SA if a duplicate is detected).

**Comment by misch42** [comment]
read the docu links that I sent.


If docu is not enough: 
https://github.com/strongswan/strongswan/discussions/196

**Reply by zephyrbl** [reply]
Dear,

Thanks but I've already read all this documents, I just don't understand how to use ./configure option, one short example will be more clear for me :)

Just for your information, I'm installing StrongSwan using the "apt install strongswan", probably I have to do this in another way?

**Comment by misch42** [comment]
On 28.04.23 12:25, cynepmann wrote:
>
> Dear,
>
> Thanks but I've already read all this documents, I just don't 
> understand how to use ./configure option, one short example will be 
> more clear for me :)
>

Did you also read the documentation how to post to this list?

**Reply by zephyrbl** [reply]
So, now I installed duplicheck plugin, enabled it in strongswan.conf but there is no difference - I still can connect from 2 devices using the same credentials.

**Comment by tobiasbrunner** [comment, marked as answer]
> uniqueids=never

This disables uniqueness checks based on identities. Change it to `uniqueids=yes` (that's actually the default).

**Reply by zephyrbl** [reply]
Dear Tobias,

I've already tried to set "yes" in uniqueid section but it doesn't work in fact, I still can connect from two devices simultaneously with the same EAP login/password.


UPD: Actually, it's working, I can connect with two devices simultaneously but only one of them will have an Internet access.

**Reply by tobiasbrunner** [reply]
> UPD: Actually, it's working, I can connect with two devices simultaneously but only one of them will have an Internet access.

Yes, the old SA will be closed if a duplicate is detected (check the log) and if an INITIAL_CONTACT notify is received on the new SA, this happens without notification on the old SA. That is, for the first device it will seem like the connection is still there (until it sends the next IKE message), it just won't work anymore because the IPsec SAs on the server are gone.
